File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/javaBuilder/chartModifier2D/CustomModifiersTheChartModifierBaseAPI.java
package com.scichart.docsandbox.examples.javaBuilder.chartModifier2D;

import androidx.annotation.NonNull;

import com.scichart.charting.model.AxisCollection;
import com.scichart.charting.modifiers.ChartModifierBase;
import com.scichart.charting.visuals.ISciChartSurface;
import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.charting.visuals.axes.AxisAlignment;
import com.scichart.charting.visuals.axes.IAxis;
import com.scichart.core.framework.IUpdateSuspender;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;

@ExampleDefinition()
public class CustomModifiersTheChartModifierBaseAPI extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) {}

    void createCustomRotateChartModifier(@NonNull SciChartSurface surface) {
        // <CreateCustomRotateChartModifier>
        class CustomRotateChartModifier extends ChartModifierBase {
            public void rotateChart() {
                if (isAttached()) {
                    final ISciChartSurface parentSurface = getParentSurface();

                    IUpdateSuspender updateSuspender = parentSurface.suspendUpdates();
                    try {
                        rotateAxes(parentSurface.getXAxes());
                        rotateAxes(parentSurface.getYAxes());
                    } finally {
                        updateSuspender.dispose();
                    }
                }
            }

            private void rotateAxes(AxisCollection axes) {
                for (int i = 0, size = axes.size(); i < size; i++) {
                    final IAxis axis = axes.get(i);
                    final AxisAlignment axisAlignment = axis.getAxisAlignment();

                    switch (axisAlignment) {
                        case Right:
                            axis.setAxisAlignment(AxisAlignment.Bottom);
                            break;
                        case Left:
                            axis.setAxisAlignment(AxisAlignment.Top);
                            break;
                        case Top:
                            axis.setAxisAlignment(AxisAlignment.Right);
                            break;
                        case Bottom:
                            axis.setAxisAlignment(AxisAlignment.Left);
                            break;
                        case Auto:
                            if(axis.isXAxis()) {
                                // Bottom
                                axis.setAxisAlignment(AxisAlignment.Left);
                            } else {
                                // Right
                                axis.setAxisAlignment(AxisAlignment.Bottom);
                            }
                    }
                }
            }
        }
        // </CreateCustomRotateChartModifier>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/chartModifier2D/CustomModifiersTheChartModifierBaseAPI.kt
package com.scichart.docsandbox.examples.kotlin.chartModifier2D

import com.scichart.charting.model.AxisCollection
import com.scichart.charting.modifiers.ChartModifierBase
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.axes.AxisAlignment
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment

@ExampleDefinition()
class CustomModifiersTheChartModifierBaseAPI : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {}

    fun createCustomRotateChartModifier(surface: SciChartSurface) {
        // <CreateCustomRotateChartModifier>
        class CustomRotateChartModifier : ChartModifierBase() {
            fun rotateChart() {
                if (isAttached) {
                    val parentSurface = parentSurface
                    val updateSuspender = parentSurface.suspendUpdates()
                    try {
                        rotateAxes(parentSurface.xAxes)
                        rotateAxes(parentSurface.yAxes)
                    } finally {
                        updateSuspender.dispose()
                    }
                }
            }

            private fun rotateAxes(axes: AxisCollection) {
                var i = 0
                val size = axes.size
                while (i < size) {
                    val axis = axes[i]
                    val axisAlignment = axis.axisAlignment
                    when (axisAlignment) {
                        AxisAlignment.Right -> axis.axisAlignment = AxisAlignment.Bottom
                        AxisAlignment.Left -> axis.axisAlignment = AxisAlignment.Top
                        AxisAlignment.Top -> axis.axisAlignment = AxisAlignment.Right
                        AxisAlignment.Bottom -> axis.axisAlignment = AxisAlignment.Left
                        AxisAlignment.Auto -> if (axis.isXAxis) {
                            // Bottom
                            axis.axisAlignment = AxisAlignment.Left
                        } else {
                            // Right
                            axis.axisAlignment = AxisAlignment.Bottom
                        }
                    }
                    i++
                }
            }
        }
        // </CreateCustomRotateChartModifier>
    }
}

